extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -167,17 +191,68 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
   }
 
   /** @private */
-  onPageNavigation_() {
-    if (this.isOnActivePage_) {
-      this.animate_();
+  setAnimateTo_() {
+    const x = parseFloat(this.element_.getAttribute('x') || 0);
+    const y = parseFloat(this.element_.getAttribute('y') || 0);
+    const zoom = parseFloat(this.element_.getAttribute('zoom') || 1);
+    const lockBounds = this.element_.hasAttribute('lock-bounds');
+
+    if (lockBounds) {
+      // Zoom must be set to calculate maxBounds.
+      this.animateTo_.zoom = zoom < 1 ? 1 : zoom;
+      const maxBounds = this.getMaxBounds_();
+      this.animateTo_.x =
+        x > 0 ? Math.min(maxBounds.left, x) : Math.max(maxBounds.right, x);
```

Comment:
`Math.max(min, Math.min(number, max))` is more performant and avoids conditional logic